### DISCUSSION TITLE
Seeing full page reloads when using localePrefix: "as-needed"

### DISCUSSION THREAD

**Discussion by Meemaw** [amannn/next-intl#738] [General]
As the title says, seeing full page reloads on client side navigations when using `localePrefix: "as-needed"`. Navigations are smooth when using `localePrefix: "always"`.

Is this expected/known issue?

**Comment by amannn** [comment]
This is definitely not intended and also something I haven't seen so far. Do you have a reproduction?

**Reply by Meemaw** [reply]
Hey, figured it out. It turns out it was related to our middleware matcher + CSP directive (upgrade-insecure-requests;) which failed some nextjs requests and triggered full reload.
